astropy/coordinates/angles.py

```diff
@@ -363,6 +363,30 @@ def do_format(val):
             result = result[()]
         return result
 
+    def _wrap_at(self, wrap_angle):
+        """
+        Implementation that assumes ``angle`` is already validated
+        and that wrapping is inplace.
+        """
+        # Convert the wrap angle and 360 degrees to the native unit of
+        # this Angle, then do all the math on raw Numpy arrays rather
+        # than Quantity objects for speed.
+        a360 = u.degree.to(self.unit, 360.0)
+        wrap_angle = wrap_angle.to_value(self.unit)
+        wrap_angle_floor = wrap_angle - a360
+        self_angle = self.view(np.ndarray)
+        # Do the wrapping, but only if any angles need to be wrapped
+        #
+        # This invalid catch block can be removed when the minimum numpy
+        # version is >= 1.19 (NUMPY_LT_1_19)
+        with np.errstate(invalid='ignore'):
+            wraps = (self_angle - wrap_angle_floor) // a360
+        if np.any(wraps != 0):
+            self_angle -= wraps*a360
+            # Rounding errors can cause problems.
+            self_angle[self_angle >= wrap_angle] -= a360
```

Comment:
Must admit not being very happy with this, but without it, one of the wrapping tests fails (it was pure luck that the previous code rounded better...)